File: src/DTO/RegisterCampDTO.ts
import {IsNotEmpty} from "class-validator";
import {ApiProperty} from "@nestjs/swagger";

export class RegisterCampDTO {
    @ApiProperty()
    @IsNotEmpty()
    firstName: string;

    @ApiProperty()
    @IsNotEmpty()
    lastName: string;

    @ApiProperty()
    @IsNotEmpty()
    email: string;

    @ApiProperty()
    @IsNotEmpty()
    city: string;

    @ApiProperty()
    @IsNotEmpty()
    phoneNumber: string;

    @ApiProperty()
    @IsNotEmpty()
    university: string;

    @ApiProperty()
    @IsNotEmpty()
    age: number;

    @ApiProperty()
    @IsNotEmpty()
    Q1: string;

    @ApiProperty()
    @IsNotEmpty()
    Q2: string;

    @ApiProperty()
    @IsNotEmpty()
    Q3: string;

    @ApiProperty()
    @IsNotEmpty()
    Q4: string;

    @ApiProperty()
    @IsNotEmpty()
    Q5: string;

    @ApiProperty()
    @IsNotEmpty()
    Q6: string;

    @ApiProperty()
    @IsNotEmpty()
    Q7: string;

    @ApiProperty()
    @IsNotEmpty()
    Q8: string;

    @ApiProperty()
    @IsNotEmpty()
    Q9: string;

    @ApiProperty()
    @IsNotEmpty()
    Q10: string;
    @ApiProperty()
    @IsNotEmpty()
    Q11: string;


}

File: src/main/java/iewa/api/DTO/Camp/RegisterCampDTO.java
package iewa.api.DTO.Camp;

import jakarta.validation.constraints.NotEmpty;
import lombok.Data;

@Data
public class RegisterCampDTO {
    private String question1;
    private String q2;
    private String q3;
    private String q4;
    private String q5;
    private String q6;
    private String q7;
    private String q8;
    private String q9;
    private String q10;
    private String q11;
    private String firstName;
    private String lastName;
    private String email;
    private String city;
    private String phoneNumber;
    private String university;
    private int age;
    private String graduated;
}